Move QScheme ops to c10

xcheng16 opened the pull request:
Stack from [ghstack](https://github.com/ezyang/ghstack):
* **#30134 Move QScheme ops to c10**

Differential Revision: [D18609716](https://our.internmc.facebook.com/intern/diff/D18609716/)

Commit:
Move QScheme ops to c10



Differential Revision: [D18609716](https://our.internmc.facebook.com/intern/diff/D18609716/)

[ghstack-poisoned]

Commit:
Update on "Move QScheme ops to c10"


Differential Revision: [D18609716](https://our.internmc.facebook.com/intern/diff/D18609716/)

[ghstack-poisoned]

Commit:
Update on "Move QScheme ops to c10"


Differential Revision: [D18609716](https://our.internmc.facebook.com/intern/diff/D18609716/)

[ghstack-poisoned]

smessmer commented on aten/src/ATen/core/jit_type.h:
```diff
@@ -1090,6 +1091,28 @@ struct CAFFE2_API GeneratorType : public Type {
+struct CAFFE2_API QschemeType : public Type {
+  static QschemeTypePtr create() {
+    return QschemeTypePtr(
+        new QschemeType()); // NOLINT(modernize-make-shared)
```
let's use make_shared here

smessmer commented on aten/src/ATen/core/jit_type.h:
```diff
@@ -1090,6 +1091,28 @@ struct CAFFE2_API GeneratorType : public Type {
   GeneratorType() : Type(TypeKind::GeneratorType) {}
 };
 
+struct QschemeType;
```
I think the naming convention is `QScheme`, not `Qscheme`

smessmer commented on aten/src/ATen/core/jit_type.h:
```diff
@@ -1090,6 +1091,28 @@ struct CAFFE2_API GeneratorType : public Type {
+    return "Qscheme";
+  }
+  static const TypeKind Kind = TypeKind::QschemeType;
+  // global singletonq
```
typo

smessmer reviewed (approved):
looks good, assuming CI passes

Commit:
Update on "Move QScheme ops to c10"


Differential Revision: [D18609716](https://our.internmc.facebook.com/intern/diff/D18609716/)

[ghstack-poisoned]

Commit:
Update on "Move QScheme ops to c10"


Differential Revision: [D18609716](https://our.internmc.facebook.com/intern/diff/D18609716/)

[ghstack-poisoned]
